nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/converters/CommonSenseConverter.java

```diff
@@ -39,51 +39,97 @@
  */
 package fish.payara.nucleus.microprofile.config.converters;
 
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.config.spi.Converter;
+
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import org.eclipse.microprofile.config.inject.ConfigProperty;
-import org.eclipse.microprofile.config.spi.Converter;
+import java.lang.reflect.Modifier;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
 
 /**
  *
  * @author steve
  */
 public class CommonSenseConverter implements Converter<Object> {
 
+    /**
+     * Return implicit converter for a class following section "Automatic Converters" of the spec, if class has
```

Comment:
Just for the record: I'd find the name `AutomaticConverter` as suggested by this comment much more explanatory.